zenoh-plugin-ros2dds/src/route_service_cli.rs

```diff
@@ -351,24 +352,46 @@ fn route_dds_request_to_zenoh(
         return;
     }
 
-    let zbuf: ZBuf = sample.into();
-    let slice = zbuf.to_zslice();
+    let z_bytes: ZBytes = sample.into();
+    let slice = Cow::from(z_bytes);
     let dds_req_buf = slice.as_ref();
+
     let is_little_endian =
         is_cdr_little_endian(dds_req_buf).expect("Shouldn't happen: sample.len >= 20");
+
     let request_id = CddsRequestHeader::from_slice(
         dds_req_buf[4..20]
             .try_into()
             .expect("Shouldn't happen: sample.len >= 20"),
         is_little_endian,
     );
 
-    // route request buffer stripped from request_id (client_id + sequence_number)
-    let mut zenoh_req_buf = ZBuf::empty();
     // copy CDR Header
-    zenoh_req_buf.push_zslice(slice.subslice(0, 4).unwrap());
+    let cdr_header = match dds_req_buf.get(0..4) {
+        Some(hdr) => hdr,
+        None => {
+            // TODO: do we want this to return or explode ?
+            tracing::error!(
+                "Sample buffer was empty, Could not ger cdr header expected bytes [0..4]",
+            );
+            return;
+        }
+    };
+
     // copy Request payload, skiping client_id + sequence_number
-    zenoh_req_buf.push_zslice(slice.subslice(20, slice.len()).unwrap());
+    let payload = match dds_req_buf.get(20..dds_req_buf.len()) {
```

Comment:
Couldn't we just use dds_req_buf[20..dds_req_buf.len()]?

Reply:
See above.